docs/blockentities/index.md

```diff
@@ -1,141 +1,239 @@
+## Storing Data
 
-### Synchronizing on LevelChunk Load
+One of the main purposes of `BlockEntity`s is to store data. Data storage on block entities can happen in two ways: directly reading and writing [NBT][nbt], or using [data attachments][dataattachments]. This section will cover reading and writing NBT directly; for data attachments, please refer to the linked article.
 
-For this you need to override
-```java
-BlockEntity#getUpdateTag(HolderLookup.Provider registries)
+Data can be read from and written to a `CompoundTag` using the `#loadAdditional` and `#saveAdditional` methods, respectively. These methods are called when the block entity is synced to disk or over the network.
 
-IBlockEntityExtension#handleUpdateTag(CompoundTag tag, HolderLookup.Provider registries)
+```java
+public class MyBlockEntity extends BlockEntity {
+    // This can be any value of any type you want, so long as you can somehow serialize it to NBT.
+    // We will use an int for the sake of example.
+    private int value;
+
+    public MyBlockEntity(BlockPos pos, BlockState state) {
+        super(MY_BLOCK_ENTITY.get(), pos, state);
+    }
+
+    // Read values from the passed CompoundTag here.
+    @Override
+    public void loadAdditional(CompoundTag tag, HolderLookup.Provider registries) {
+        super.loadAdditional(tag, registries);
+        // Will default to 0 if absent. See the NBT article for more information.
+        this.value = tag.getInt("value");
+    }
+
+    // Save values into the passed CompoundTag here.
+    @Override
+    public void saveAdditional(CompoundTag tag, HolderLookup.Provider registries) {
+        super.saveAdditional(tag, registries);
+        tag.putInt("value", this.value);
+    }
+}
 ```
 
-The first method collects the data that should be sent to the client while the second one processes that data. If your `BlockEntity` doesn't contain much data, you might be able to use the methods out of the [Storing Data within your `BlockEntity`][storing-data] section.
+In both methods, it is important that you call super, as that adds basic information such as the position. The tag names `id`, `x`, `y`, `z`, `NeoForgeData` and `neoforge:attachments` are reserved by the super methods, and as such, you should not use them yourself.
 
-:::caution
-Synchronizing excessive/useless data for block entities can lead to network congestion. You should optimize your network usage by sending only the information the client needs when the client needs it. For instance, it is more often than not unnecessary to send the inventory of a block entity in the update tag, as this can be synchronized via its [`AbstractContainerMenu`][menu].
+:::info
+It is expected that Mojang will adapt the [Data Components][datacomponents] system to also work with block entities sometime during the next few updates. Once that happens, both saving to NBT and data attachments will be removed in favor of data components.
```

Comment:
I would not jump to this conclusion yet. We'll get there when and if it actually happens.